android/src/main/java/org/mozilla/mozstumbler/client/mapview/maplocation/UserPositionUpdateManager.java

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.mozstumbler.client.mapview.maplocation;
+
+import android.content.Context;
+import android.location.GpsSatellite;
+import android.location.GpsStatus;
+import android.location.Location;
+import android.location.LocationManager;
+import android.widget.Toast;
+import java.lang.ref.WeakReference;
+import org.mozilla.mozstumbler.client.MainApp;
+import org.mozilla.mozstumbler.client.mapview.MapFragment;
+import org.mozilla.mozstumbler.service.AppGlobals;
+import org.mozilla.mozstumbler.service.core.logging.ClientLog;
+import org.mozilla.mozstumbler.service.stumblerthread.scanners.GPSScanner;
+import org.mozilla.mozstumbler.svclocator.services.log.LoggerUtil;
+
+
+public class UserPositionUpdateManager {
+    private static final String LOG_TAG = LoggerUtil.makeLogTag(UserPositionUpdateManager.class);
+    private final WeakReference<MapFragment> mMapFragment;
+    private LocationManager mLocationManager;
+    private final MultiSourceLocationListener mMultiSourceLocationListener;
+
+    private final GpsStatus.Listener mSatelliteListener = new GpsStatus.Listener() {
+        public void onGpsStatusChanged(int event) {
+            if (event == GpsStatus.GPS_EVENT_SATELLITE_STATUS && mLocationManager != null) {
+                GpsStatus gpsStatus = mLocationManager.getGpsStatus(null);
+                Iterable<GpsSatellite> sats = gpsStatus.getSatellites();
+                int satellites = 0;
+                int fixes = 0;
+                for (GpsSatellite sat : sats) {
+                    satellites++;
+                    if (sat.usedInFix()) {
+                        fixes++;
+                    }
+                }
+                if (mMapFragment.get() != null) {
+                    mMapFragment.get().updateGPSInfo(satellites, fixes);
+                }
+
+                if (fixes < GPSScanner.MIN_SAT_USED_IN_FIX) {
```

Comment:
Can we push `mSatelliteListener`, the registration of `addGpsStatusListener` and this `if` block into the multisource location listener - possibly down into the GPS update listener?  It doesn't make sense that we have GPS code up in the UserPositionUpdateManager when we already have an explcit GPS listener class that is assisting here.